### PR TITLE
Web Custom Avatar

### PR DESCRIPTION
**Pull request by Rikimbili**
Had to bring back useIsClient to workaraound localstorage and hydration issues

### CODE REVIEW

File: web/src/requests/mutate.ts [hunk 1]
```diff
@@ -31,6 +32,18 @@ export async function register(name: string, email: string, password: string) {
...
+    "/api/user/update",
+    {
+      userId: getAuthSession().userId,
+      image,
+      userName,
+    },
+    "POST"
+  );
+}
+
```
Comment (rob-3) [inline web/src/requests/mutate.ts:46]: Please mirror this over in `mobile`!

File: web/src/components/Surfaces/ProfileImage.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        >
+          {!isLoading ? (
+            <>
+              <input
+                type={"file"}
+                onChange={onImageChange}
+                title={" "}
+                aria-label={"Set Profile Image"}
+                className={
+                  "absolute z-10 h-full w-full  cursor-pointer opacity-0"
```
Comment (rob-3) [inline web/src/components/Surfaces/ProfileImage.tsx:36]: Double space in this line (doesn't matter, but just noticed it)

File: web/src/pages/api/user/update.ts [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    .catch((err) => {
+      console.error(err);
+      res.status(500).json({ error: "Internal database error" });
+    });
+}
+
+async function uploadImage(image: string, res: NextApiResponse) {
+  // Resize and compress the image before uploading it to imgur
+  // Sharp will resize  both images and animated gifs
+  const uri = image.split(";base64,").pop() as string;
```
Comment (rob-3) [inline web/src/pages/api/user/update.ts:82]: I would use `.at(-1)` instead of `.pop()` unless you have a good reason I don't know about.

**Review by rob-3 — CHANGES_REQUESTED** [review]

File: web/src/requests/mutate.ts (hunk 1 above)
Comment (Rikimbili) [reply web/src/requests/mutate.ts:46]: oh true, I keep forgetting 💀

File: web/src/components/Surfaces/ProfileImage.tsx (hunk 2 above)
Comment (Rikimbili) [reply web/src/components/Surfaces/ProfileImage.tsx:36]: It is not longer there on the latest branch im working on

**Review by rob-3 — APPROVED** [review]

**Pull request merged**